automotive/odcantools/libodcantools/include/CANMessage.h

```diff
@@ -0,0 +1,74 @@
+#ifndef CAN_MESSAGE_H_
+#define CAN_MESSAGE_H_
+
+#include <automotivedata/generated/automotive/GenericCANMessage.h>
+
+#include <vector>
+#include <map>
+
+namespace automotive {
+    namespace odcantools {
+        class CANMessage;
+        
+        class CANSignal {
+            friend class CANMessage;
+            
+            public:
+                CANSignal(const uint8_t, const uint8_t, const string, const string, const double, const double, const double, const double);
```

Comment:
@fgiaimo Can we have `const Type&` here to avoid unnecessary value copying (CAN message handling is typically executed with higher frequencies and saving costs by avoiding call by value is preferred)